src/offset_date_time.rs

```diff
@@ -729,6 +729,77 @@ impl OffsetDateTime {
     }
 
     // endregion: checked arithmetic
+
+    // region: saturating arithmetic
+    /// Computes `self + duration`, saturating value on overflow.
+    ///
+    /// ```
+    /// # use time::ext::NumericalDuration;
+    /// # use time::macros::datetime;
+    ///
+    /// assert_eq!(
+    ///     datetime!(-999999 - 01 - 01 00:00 +10:00).saturating_add((-2).days()),
+    ///     datetime!(-999999 - 01 - 01 00:00 +10:00)
+    /// );
+    ///
+    /// assert_eq!(
+    ///     datetime!(+999999 - 12 - 31 23:59:59.999_999_999 +10:00).saturating_add(2.days()),
+    ///     datetime!(+999999 - 12 - 31 23:59:59.999_999_999 +10:00)
+    /// );
+    ///
+    /// assert_eq!(
+    ///     datetime!(2019 - 11 - 25 15:30 +10:00).saturating_add(27.hours()),
+    ///     datetime!(2019 - 11 - 26 18:30 +10:00)
+    /// );
+    /// ```
+    pub const fn saturating_add(self, duration: Duration) -> Self {
+        if let Some(datetime) = self.checked_add(duration) {
+            datetime
+        } else if duration.is_negative() {
+            PrimitiveDateTime::MIN
+                .assume_utc()
+                .replace_offset(self.offset)
```

Comment:
Okay, I'll replace it with `to_offset` then.

Reply:
Hm...  Nope. It should be `replace_offset` because I'm building `datetime!(-999999 - 01 - 01 00:00 OFFSET)` here. With `to_offset` it will be something else.  Assuming `OffsetDateTime`'s minimum value is `-999999 - 01 - 01 00:00 OFFSET` and maximum is `999999 - 12 - 31 23:59:59.999999999 OFFSET`.